sourcegraph/learn
# Move interfaces and types to dedicated folders

### What should this PR do?
Resolves [DEVED-74](https://sourcegraph.atlassian.net/browse/DEVED-74) by moving `interfaces` and `types` to dedicated folders. The thinking here is that if we scope the `util` folder to be only shared utility functions, it may make the project's organization easier to reason about.

### Why are we making this change?
- We want to have things organized and scoped in such a way that folks can onboard to the project as quickly as possible.

### What are the acceptance criteria? 
- Everything should work as it currently does in prod and the organizational changes make sense.

### How should this PR be tested?
- Check out the branch and ensure everything works as it currently does in pod & sanity check organizational changes.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
